### PR TITLE
Properly dispose the SymbolSearchEngine

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Noticed while investigating post-VS shutdown issues with Dave.

### CODE REVIEW

File: src/Features/Core/Portable/SymbolSearch/SymbolSearchUpdateNoOpEngine.cs [hunk 1]
```diff
@@ -13,6 +13,11 @@ internal sealed class SymbolSearchUpdateNoOpEngine : ISymbolSearchUpdateEngine
 {
     public static readonly SymbolSearchUpdateNoOpEngine Instance = new();
 
+    public void Dispose()
+    {
+        // Nothing to do for the no-op version.
+    }
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SymbolSearch/SymbolSearchUpdateNoOpEngine.cs:19]: Note: the actual Dispose we care about *already* exists, but is currently not called through any codepaths.  It is here: https://github.com/dotnet/roslyn/blob/9b6dd878170636c4a7771aae644aede20123213b/src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngineProxy.cs#L16

What thsi PR does is hook things up to ensure that it is called.  To ensure that we actually dispose the RemoteServiceConnection instance we're holding onto.

**Comment by CyrusNajmabadi** [comment]
@ToddGrun ptal

File: src/VisualStudio/Core/Def/SymbolSearch/VisualStudioSymbolSearchService.cs [hunk 2]
```diff
@@ -69,7 +65,20 @@ internal partial class VisualStudioSymbolSearchService : AbstractDelayStartedSer
...
     {
         _serviceProvider = serviceProvider;
-        _installerService = workspace.Services.GetService<IPackageInstallerService>();
+        _installerService = workspace.Services.GetRequiredService<IPackageInstallerService>();
+    }
+
+    public void Dispose()
+    {
+        ISymbolSearchUpdateEngine? updateEngine;
+        using (_gate.DisposableWait())
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/SymbolSearch/VisualStudioSymbolSearchService.cs:74]: indeed.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**